part2/decomposition.py

```
@@ -127,7 +129,10 @@ def svd_decompose(
 
     # tìm trị riêng của A^T * A và vector riêng của V
     try:
-        lambdas, V = jacobi_eigenvalues(ATA, max_iterations)
+        lambdas, V = jacobi_eigenvalues(
+            ATA,
+            error_tolerance=1e-10,
+            max_iterations=max_iterations)
     except ValueError as e:
         raise RuntimeError(f"SVD failed due to Jacobi eigenvalue error: {e}")
```

Ruff (0.15.9)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)